test/testing/test_services.py

```diff
@@ -352,15 +373,200 @@ def test_entrypoint_waiter(container_factory, rabbit_config):
+        name = "service"
+
+        @event_handler('srcservice', 'eventtype')
+        def handle_event(self, msg):
+            return msg
+
+    container = container_factory(Service, rabbit_config)
+    container.start()
+
+    results = []
+
+    def cb(worker_ctx, res, exc_info):
+        results.append((res, exc_info))
+        if len(results) == 2:
+            return True
+        return False
+
+    dispatch = event_dispatcher(rabbit_config)
+    with entrypoint_waiter(container, 'handle_event', callback=cb):
+        dispatch('srcservice', 'eventtype', "msg1")
+        dispatch('srcservice', 'eventtype', "msg2")
+
+    assert results == [("msg1", None), ("msg2", None)]
+
+
+def test_entrypoint_waiter_wait_for_specific_result(
+    container_factory, rabbit_config, spawn_thread
+):
+
+    class Service(object):
+        name = "service"
+
+        @event_handler('srcservice', 'eventtype')
+        def handle_event(self, msg):
+            return msg
+
+    container = container_factory(Service, rabbit_config)
+    container.start()
+
+    target = 5
+
+    def cb(worker_ctx, res, exc_info):
+        if res == target:
+            return True
+
+    def increment_forever():
+        dispatch = event_dispatcher(rabbit_config)
+        for count in itertools.count():
+            dispatch('srcservice', 'eventtype', count)
+            time.sleep()  # force yield
+
+    with entrypoint_waiter(container, 'handle_event', callback=cb) as result:
+        spawn_thread(increment_forever)
+
+    assert result.get() == target
+
+
+def test_entrypoint_waiter_wait_until_called_with_argument(
+    container_factory, rabbit_config, spawn_thread
+):
+
+    class Service(object):
+        name = "service"
+
+        @event_handler('srcservice', 'eventtype')
+        def handle_event(self, msg):
+            return msg
+
+    container = container_factory(Service, rabbit_config)
+    container.start()
+
+    target = 5
+
+    def cb(worker_ctx, res, exc_info):
+        if worker_ctx.args == (target,):
+            return True
+
+    def increment_forever():
+        dispatch = event_dispatcher(rabbit_config)
+        for count in itertools.count():
+            dispatch('srcservice', 'eventtype', count)
+            time.sleep()  # force yield
+
+    with entrypoint_waiter(container, 'handle_event', callback=cb) as result:
+        spawn_thread(increment_forever)
+
+    assert result.get() == target
+
+
+def test_entrypoint_waiter_wait_until_raises(
+    container_factory, rabbit_config, spawn_thread
+):
+    threshold = 5
+
+    class TooMuch(Exception):
+        pass
+
+    class Service(object):
+        name = "service"
+
+        @event_handler('srcservice', 'eventtype')
+        def handle_event(self, msg):
+            if msg > threshold:
+                raise TooMuch(msg)
+            return msg
+
+    container = container_factory(Service, rabbit_config)
+    container.start()
+
+    def cb(worker_ctx, res, exc_info):
```

Comment:
what are your thoughts in general on 

```
def foo():
    if condition:
        return True
    # possibly
    else:
        return False
```

vs

```
def foo():
    return condition
```

you seem to be favouring the former. for readability?